### PR TITLE
add database support for decimal course credits

### PR DESCRIPTION
**Pull request by ramsays**
refs gradebook/engineering#106

Based on user feedback to allow for decimal course credits.

**Meta**
Reviewers: 1

### CODE REVIEW

File: lib/database/migrations/000010-allow-decimal-course-credits.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// @ts-check
```
Comment (vikaspotluri123) [inline lib/database/migrations/000010-allow-decimal-course-credits.js:1]: Can we update the filename to reflect the schema change (`convert-course-credits-column-to-float`)

File: lib/database/migrations/000010-allow-decimal-course-credits.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import createAlterColumn from '../migration-utils/alter-column.js';
+
+const {up, down} = createAlterColumn({
+	table: 'courses',
+	columnName: 'credit_hours',
+	previousSchema: {
+		type: 'tinyint', nullable: false, unsigned: true, validations: {between: [0, MAX_CREDITS_IN_COURSE]},
+	},
+	currentSchema: {type: 'float', nullable: false, validations: {between: [0, MAX_CREDITS_IN_COURSE]}},
+	context: 'Allow for decimal course credits',
```
Comment (vikaspotluri123) [inline lib/database/migrations/000010-allow-decimal-course-credits.js:12]: Same here

**Comment by vikaspotluri123** [comment]
Other than the above comments, LGTM!

 * sqlite3 isn't really typed so the migration is more for schema accuracy than data consistency
 * Tested against sqlite3 and confirmed that up/down works with decimal hours and no data loss
 * Tested against MariaDB and confirmed that up/down works with decimal hours, but the decimal is truncated on a down

File: lib/database/migrations/000010-allow-decimal-course-credits.js (hunk 2 above)
Comment (ramsays) [reply lib/database/migrations/000010-allow-decimal-course-credits.js]: Yep, made them consistent 👍

**Review by vikaspotluri123 — APPROVED** [review]

**Pull request merged**